### PR TITLE
Enhance library contribution docs

### PR DESCRIPTION
**Pull request by gavin-norman-sociomantic**

### CODE REVIEW

**Comment by ben-palmer-sociomantic** [comment]
LGTM

**Comment by gavin-norman-sociomantic** [comment]
Updated first commit as discussed.

File: doc/library-contributor.rst [hunk 1]
```diff
@@ -10,6 +10,18 @@ Neptune-versioned libraries should read
 
 .. contents::
 
+Submitting Pull Requests
+------------------------
+
+When making a PR to a Neptune-versioned library, the choice of which branch to
+submit a PR against is important. A branch should be chosen as follows:
+* If your change is a bug fix, the PR should be submitted against the oldest
+  supported minor branch.
```
Comment (mathias-baumann-sociomantic) [inline doc/library-contributor.rst]: This is formatted wrong: https://github.com/gavin-norman-sociomantic/neptune/blob/5ee42034c21ec31a25e6f289451c56ac4fc74a80/doc/library-contributor.rst#id1

File: doc/library-contributor.rst [hunk 2]
```diff
@@ -16,10 +16,10 @@ Release Notes
...
 ........................
 
-When making a commit to a Neptune-versioned library, API-affected changes should
-be noted in a file in the ``relnotes`` folder, as follows. When the
-corresponding branch is released, the files in that folder will be collated to
-form the notes for the release.
+When making a commit to a Neptune-versioned library, changes that are not purely
+internal implementation details should be noted in a file in the ``relnotes``
+folder, as follows. When the corresponding branch is released, the files in that
+folder will be collated to form the notes for the release.
```
Comment (mathias-baumann-sociomantic) [inline doc/library-contributor.rst]: I found the mentioning of API very clear compared to this

**Review by mathias-baumann-sociomantic — CHANGES_REQUESTED** [review]

File: doc/library-contributor.rst (hunk 1 above)
Comment (gavin-norman-sociomantic) [reply doc/library-contributor.rst]: Bloody rst.

File: doc/library-contributor.rst (hunk 2 above)
Comment (gavin-norman-sociomantic) [reply doc/library-contributor.rst]: Yeah, I'll remove this change. I think "API-affecting" is totally clear. (There was some discussing around whether "extending" is covered by "affecting". I believe it is.)

**Comment by gavin-norman-sociomantic** [comment]
Updated.

**Review by mathias-baumann-sociomantic — APPROVED** [review]

**Pull request merged**